BSP-18 : Associate a contract to a user

maissanouhaud opened the pull request:
- Working on the creation of an inventory contract, all the types are implemented, have to fix some issues with warehouse types and typescript notation

Nysnaa committed:
BSP-18: Create inventory contract and assign items

- Add private handlers to create inventory and assign items
- Refactor to public/private handlers and services

podefr commented on src/config.ts:
```diff
@@ -41,10 +44,34 @@ if (
     );
 }
 
+if (!INVENTORY_TEZOS_SECRET_KEY) {
```
the secret key and public key hash will be the same whether we deploy an inventory or warehouse contract, it's just a way for the user (owner of both contract) to authenticate with the tezos node. So I think we can remove those 2 environment variables

podefr commented on src/config.ts:
```diff
@@ -41,10 +44,34 @@ if (
+    !validateAddress(INVENTORY_CONTRACT_ADDRESS)
+) {
+    throw new Error(
+        "Please provide a valid KT1 address to access the INVENTORY via INVENTORY_CONTRACT_ADDRESS or leave blank for the contract to be automatically deployed"
```
so here, we won't be deploying an inventory contract upon startup, contrary to warehouse.
there is only one warehouse contract for the whole platform, but there is at least one inventory per user that we create.
So the inventory contract isn't deployed when the application starts, only when the related handler is called

podefr commented on src/handlers/inventory.ts:
```diff
@@ -0,0 +1,72 @@
+
+export const inventoryHandlers: NatsHandler[] = [
+    [
+        "tokenization-service_assign_item",
```
we should refactor this service to follow the same conventions as the item-store: 

- tokenization-service is prefixed when registering the handlers
- the logic for calling the smart contract entrypoints should be moved into a services/inventoryContract.ts file

podefr committed:
BSP-28: Add public handlers for managing the warehouse and the inventory

- Add warehouse public handlers to tokenize and update items
- Add inventory public handlers to assign and update items
- Add postman collection for testing
